src/main/java/ure/UREAreaGenerator.java

```diff
@@ -0,0 +1,9 @@
+package ure;
+
+import java.util.Random;
+
+public abstract class UREAreaGenerator {
+    protected static Random random = new Random();
+
+    public abstract UREArea generate(URETerrainCzar terrainCzar, int totalSteps, int sizeX, int sizeY);
```

Comment:
i don't think having a generic generate() method with the same parameters for every child is gonna work out, see below commentary